File: 0869-Reordered Power of 2/0869-Reordered Power of 2.cpp
class Solution {
public:
    bool reorderedPowerOf2(int N) {
        int result = countDigits(N);
        for (int i = 0; i < 32; ++i) {
            if (result == countDigits(1 << i)) {
                return true;
            }
        }
        return false;
    }
    
private:
    int countDigits(int N) {
       int result = 0;
        while (N) {
            result += pow(10, N % 10);
            N /= 10;
        }
        return result;
    }

};

File: 0801-0900/0869-Reordered Power of 2/0869-Reordered Power of 2.py
class Solution:
    def reorderedPowerOf2(self, N: int) -> bool:
        def countDigits(N):
            result = 0
            while N:
                result += 10 ** (N % 10)
                N //= 10
            return result
        result = countDigits(N)
        return any(result == countDigits(1 << i) for i in range(31))


import collections
class Solution2:
    def reorderedPowerOf2(self, N: int) -> bool:
        curr = collections.Counter(str(N))
        return any(curr == collections.Counter(str(1 << i)) for i in range(31))
